app/renderer/components/menubar.js

```diff
@@ -102,193 +100,69 @@ class Menubar extends ImmutableComponent {
-   */
-  getContextMenuItemBounds () {
-    const selected = document.querySelectorAll('.contextMenuItem.selectedByKeyboard')
-    if (selected.length > 0) {
-      return selected.item(selected.length - 1).getBoundingClientRect()
-    }
-    return null
-  }
-  /**
-   * Returns index for the active / focused menu.
-   */
-  get currentIndex () {
-    return this.props.selectedIndex[this.props.selectedIndex.length - 1]
-  }
-  /**
-   * Upper bound for the active / focused menu.
-   */
-  get maxIndex () {
-    return this.getMenuByIndex().size - 1
-  }
-  /**
-   * Returns true is current state is inside a regular menu.
-   */
-  get hasMenuSelection () {
-    return this.props.selectedIndex.length > 1
-  }
-  /**
-   * Returns true if current state is inside a submenu.
-   */
-  get hasSubmenuSelection () {
-    return this.props.selectedIndex.length > 2
-  }
-  /**
-   * Fetch menu based on selected index.
-   * Will navigate children to find nested menus.
-   */
-  getMenuByIndex (parentItem, currentDepth) {
-    if (!parentItem) parentItem = this.getTemplateItemsOnly()
-    if (!currentDepth) currentDepth = 0
-
-    const selectedIndices = this.props.selectedIndex.slice(1)
-    if (selectedIndices.length === 0) return parentItem
-
-    const submenuIndex = selectedIndices[currentDepth]
-    const childItem = parentItem.get(submenuIndex)
-
-    if (childItem && childItem.get('submenu') && currentDepth < (selectedIndices.length - 1)) {
-      return this.getMenuByIndex(childItem.get('submenu'), currentDepth + 1)
-    }
-
-    return parentItem
-  }
 
   onKeyDown (e) {
     const selectedIndex = this.props.selectedIndex
+    const template = this.props.template
 
-    if (!selectedIndex || !this.props.template) return
+    if (!template) return
 
     switch (e.which) {
-      case keyCodes.ENTER:
-        e.preventDefault()
-        const selectedLabel = this.getMenuByIndex().getIn([this.currentIndex, 'label'])
-        windowActions.clickMenubarSubmenu(selectedLabel)
-        windowActions.resetMenuState()
-        break
-
       case keyCodes.LEFT:
       case keyCodes.RIGHT:
-        e.preventDefault()
-
-        // Left arrow inside a submenu
-        // <= go back one level
-        if (e.which === keyCodes.LEFT && this.hasSubmenuSelection) {
-          const newIndices = selectedIndex.slice()
-          newIndices.pop()
-          windowActions.setSubmenuSelectedIndex(newIndices)
-
-          let openedSubmenuDetails = this.props.contextMenuDetail.get('openedSubmenuDetails')
-            ? this.props.contextMenuDetail.get('openedSubmenuDetails')
-            : new Immutable.List()
-          openedSubmenuDetails = openedSubmenuDetails.pop()
-
-          windowActions.setContextMenuDetail(this.props.contextMenuDetail.set('openedSubmenuDetails', openedSubmenuDetails))
-          break
+        const currentContextMenuItem = windowStore.getState().getIn(['ui', 'contextMenu', 'selectedIndex'])
```

Comment:
Instead of getting like this, could you remove the reference to windowStore and then bind currentContextMenuItem as a property in `main.js`?
https://github.com/brave/browser-laptop/blob/master/js/components/main.js#L938